### PR TITLE
Add support for (U)Int32 and append the 'L' suffix to all Int64 values

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Language and Standard Library
- Add support for (U)Int32.
- Append the 'L' suffix to all Int64 values
```

### CODE REVIEW

File: backend/testfiles/execution/stdlib/int8.dark [hunk 1]
```diff
@@ -74,12 +82,37 @@ PACKAGE.Darklang.Stdlib.Int8.subtract_v0 0y 1y = -1y
...
 PACKAGE.Darklang.Stdlib.Int8.multiply_v0 8y 8y = 64y
 PACKAGE.Darklang.Stdlib.Int8.multiply_v0 1y 0y = 0y
 
+PACKAGE.Darklang.Stdlib.Int8.multiply_v0 64y 2y = Builtin.Test.derrorMessage
+  "Out of range"
+
+PACKAGE.Darklang.Stdlib.Int8.multiply_v0 -128y -1y = Builtin.Test.derrorMessage
+  "Out of range"
+
+PACKAGE.Darklang.Stdlib.Int8.power_v0 2y 3y = 8y
```
[the remark this answers is not in the document]
Comment (pbiggar) [reply backend/testfiles/execution/stdlib/int8.dark:100]: And something similar for the other sizes

File: packages/darklang/stdlib/int8.dark [hunk 2]
```diff
@@ -47,16 +47,13 @@ module Darklang =
...
-        else if ``base`` == 1y then
-          1y
-        else if ``base`` == -1y && (Builtin.Int8.``mod`` exponent 2y == 0y) then
-          1y
-        else if ``base`` == -1y then
-          -1y
-        else
-          Builtin.Int8.power ``base`` exponent
+        match ``base``, exponent with
+        | 0y, 0y -> 1y
```
Comment (pbiggar) [inline packages/darklang/stdlib/int8.dark:51]: I think this is much less clear than the `if` statements that were there before.

**Review by pbiggar — CHANGES_REQUESTED** [review]
Looks great! There's a few small changes and should be good.

File: backend/src/BuiltinExecution/Libs/Json.fs [hunk 3]
```diff
@@ -495,6 +501,56 @@ let parse
...
         raiseCantMatchWithType TUInt16 j pathSoFar |> Ply
 
+    | TInt32, JsonValueKind.Number ->
+      let mutable i32 = 0
+      let mutable ui32 = 0ul
+      let mutable d = 0.0
+
+      if j.TryGetInt32(&i32) then
+        DInt32 i32 |> Ply
+      else if j.TryGetUInt32(&ui32) then
```
[the remark this answers is not in the document]
Comment (OceanOak) [reply backend/src/BuiltinExecution/Libs/Json.fs:511]: I think since we are handling UInt32 separately we don't need this check. I'll remove it

**Pull request merged**